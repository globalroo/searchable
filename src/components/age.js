import React from "react";
import { getStandardDateFormat } from "src/helpers/date";
import { differenceInYears } from "date-fns";
import Alive from "@material-ui/icons/Accessibility";
import Dead from "@material-ui/icons/MoodBad";

export const Age = ({ birthday, deathday }) => {
	const birthDate = new Date(birthday);
	const toDay = deathday ? deathday : new Date();
	return (
		<div style={styles.container}>
			{deathday ? <Dead /> : <Alive />}
			{getStandardDateFormat(birthDate)}
			{" - "}
			{deathday && getStandardDateFormat(toDay)} ({differenceInYears(toDay, birthDate)} years
			old)
		</div>
	);
};

const styles = {
	container: {
		display: "flex",
		flexDirection: "row",
		alignItems: "center",
		fontSize: "smaller"
	}
};
