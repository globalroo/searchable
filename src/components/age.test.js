import React from "react";
import { render, getByText } from "react-testing-library";
import { Age } from "./age";

describe("Movie Detail tests", () => {
	it("Renders an Age component", () => {
		const { container } = render(<Age birthday={"1984/7/22"} />);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("Should show the correct age till 'now' if still alive", () => {
		const { container } = render(<Age birthday={"1984/7/22"} />);
		const found = getByText(container, /34 years old/);
		expect(found).toBeTruthy();
	});

	it("Should show the correct age till 'death day if deceased", () => {
		const { container } = render(<Age birthday={"1984/7/22"} deathday={"2004/7/22"} />);
		const found = getByText(container, /20 years old/);
		expect(found).toBeTruthy();
	});
});
