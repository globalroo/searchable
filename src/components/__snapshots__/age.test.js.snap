// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Detail tests Renders an Age component 1`] = `
<div
  style="display: flex; flex-direction: row; align-items: center; font-size: smaller;"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root-1"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 2c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2zm9 7h-6v13h-2v-6h-2v6H9V9H3V7h18v2z"
    />
  </svg>
  Sunday 22nd July 1984
   - 
   (
  34
   years old)
</div>
`;
